docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,106 @@
+        * On the monthly Firefox merge date:
+          * Create a `release_v[N]` branch of of main
+          * Create a GitHub release for that branch
+          * Bump the major version number in main and start a new changelog header.
+    * Phase 2: Automate nightly merges of `main` into consumer applications (March 2023 - May 2023)
+      * Whenever a new application-services version is released (including nightly, beta, release, and ESR versions).
+        * For each consumer application (including all mobile apps and Desktop):
+            * Create a PR to take the new application-services version.
+            * Set up an auto-merge system where if the test suite passes we automatically merge the PR.
+            * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
+    * All application-services changes that are breaking for consumers should have corresponding PRs in the consumer repos that have been reviewed and approved.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+
+## Decision Outcome
+
+**(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+
+## Pros and Cons of the Options
+
+### Keep our current release process
+
+  * Good, because it doesn't require changes to our workflow
+
+### Move to a rapid release process where changes in app-services are frequently merged into consumer applications
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Bad, because we need implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.
+  * Good, because this avoids the current state of affairs where our `main` branch is not ready to be taken by the applications.  This means that a) our current code is not getting used and tested and b) it's very complicated to get other changes to that application, including bugfixes.
+  * Bad, because breaking changes require the corresponding PRs to be approved from all other teams before they can be merged into application-services.
+  * Good, because this is how things will work in a monorepo world.  Adjusting our workflow now prepares us for the future and will give us insights that may help guide the monorepo migration.
+  * Good, because our version numbers will align more closely with the rest of Mozilla.
+  * Bad, because our version numbers will no longer follow semver.
```

Comment:
Feels like this is not necessarily bad imo -- semver is good for libraries with a variety of consumers that could possibly depend on different versions of a-s (and decide upgrade paths). Where we are essentially a module in various Mozilla products and thus it's better (as you mentioned in the point above) that we're aligning with how we're actually used

Reply:
Agreed.  I'll just remove this item, I think the note that we're breaking with semver is all we need.